tests/virt_autotest/guest_upgrade_run.pm

```diff
@@ -119,21 +119,15 @@ sub run {
     my $self = shift;
     my $timeout = get_var('MAX_TEST_TIME', '36000') + 10;
     my $upload_log_name = 'guest-upgrade-logs';
-    if (is_s390x) {
-        #ues die_on_timeout=> 0 as workaround for s390x test during call script_run, refer to poo#106765
-        script_run("echo \"Debug info: max_test_time is $timeout\"", die_on_timeout => 0);
-    } else {
-        script_run("echo \"Debug info: max_test_time is $timeout\"");
-    }
+    script_run("timeout 20 echo \"Debug info: max_test_time is $timeout\"");
```

Comment:
This does not seem equivalent. S390 needs to continue when timeout happens.

Reply:
I think `enter_cmd()` is ok for both x86 and s390. @guoxuguang  ?

Reply:
We can do

```suggestion
    if (is_s390x) {
        # workaround for s390x test during call script_run, refer to poo#106765
        enter_cmd "echo \"Debug info: max_test_time is $timeout\"; echo DONE > /dev/$serialdev";
        wait_serial 'DONE', timeout => 30;
    } else {
        script_run("echo \"Debug info: max_test_time is $timeout\"");
    }
```

or also

```suggestion
    enter_cmd "echo \"Debug info: max_test_time is $timeout\"; echo DONE > /dev/$serialdev";
    my $ret = wait_serial 'DONE', timeout => 30;
    die "Timeout exceeded on wait_serial" unless $ret or is_s390x;
```

which would both be mostly equivalent. If you still want to ignore timeouts but same for x86_64 and s390x then we just leave out the line `die …` which waits until "DONE" is found or timeout expires and then continues.

Reply:
Please see my updated version.

Reply:
The first option seems easier to understand :)

Reply:
Can you still accept the second proposal which is the version I implemented? :) I prefer it because it focuses on what should be the normal behaviour and not extraordinary error handling with less lines and less indendation.

Reply:
> I think `enter_cmd()` is ok for both x86 and s390. @guoxuguang ?

I just confirm that `enter_cmd()` is workable with s390x.

Reply:
I understood the question from @Julie-CAO was if we should just use the same approach for x86_64 and s390x. But that could still mean to die on timeout or ignore the timeout. I suggest to keep the behaviour mostly unchanged for this PR to not make it too hard to review. You can still follow-up with harmonizing the approaches for s390x && x86_64 afterwards.

Reply:
Sure, make sense. Thanks. LGTM

Reply:
@okurz I am fine with option 2 if you insist on it. After this change , I think it is good to approve from vt code regard.

Reply:
@okurz  You forgot to remove line 122

```
   script_run("timeout 20 echo \"Debug info: max_test_time is $timeout\"");
```

After this, I think it is good to merge.

Reply:
ok, done